experimental/oak_baremetal_loader/src/lookup.rs

```diff
@@ -0,0 +1,53 @@
+//
+// Copyright 2022 The Project Oak Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+use hashbrown::HashMap;
+use oak_baremetal_communication_channel::schema;
+use std::fs;
+
+pub fn encode_lookup_data<'a>(
+    mut data: HashMap<Vec<u8>, Vec<u8>>,
+) -> oak_idl::utils::Message<schema::LookupData<'a>> {
+    let mut builder = oak_idl::utils::MessageBuilder::default();
+    let entries: Vec<flatbuffers::WIPOffset<schema::LookupDataEntry>> = data
+        .drain()
+        .map(|(key, value)| {
+            let key = builder.create_vector::<u8>(&key);
+            let value = builder.create_vector::<u8>(&value);
+            schema::LookupDataEntry::create(
+                &mut builder,
+                &schema::LookupDataEntryArgs {
+                    key: Some(key),
+                    value: Some(value),
+                },
+            )
+        })
+        .collect();
+    let items = builder.create_vector(&entries);
```

Comment:
There's a bit of reallocation happening here, which is not super efficient. The file is loaded in memory as a byte array, then parsed and reallocated into a hashmap of entries, before finally being reallocated again when entered into the flatbuffer struct. 

Ideally we would read the file as a stream, and on the fly directly assemble the flatbuffer message. That doesn't seem possible atm though, since the flatbuffer builder's rust API requires  knowledge of the number of elements when creating a new vector. 

Anyways, I'm not sure if it's a priority to fully optimize this at this stage, given that this logic is basically just used for testing at this stage.

Reply:
I think this is fine for now. Seeing that we are not fully committed to using flatbuffers as an IDL in the long run, optimising at this point seems like potentially wasted effort.